core/src/main/java/com/graphhopper/routing/weighting/TurnWeighting.java

```diff
@@ -89,15 +89,14 @@ public long calcMillis(EdgeIteratorState edgeState, boolean reverse, int prevOrN
         if (prevOrNextEdgeId == EdgeIterator.NO_EDGE)
             return millis;
 
-        // TODO for now assume turn costs are returned in milliseconds?
         // should we also separate weighting vs. time for turn? E.g. a fast but dangerous turn - is this common?
-        long turnCostsInMillis;
+        long turnCostsInSeconds;
         if (reverse)
-            turnCostsInMillis = (long) calcTurnWeight(edgeState.getEdge(), edgeState.getBaseNode(), prevOrNextEdgeId);
+            turnCostsInSeconds = (long) calcTurnWeight(edgeState.getEdge(), edgeState.getBaseNode(), prevOrNextEdgeId);
         else
-            turnCostsInMillis = (long) calcTurnWeight(prevOrNextEdgeId, edgeState.getBaseNode(), edgeState.getEdge());
+            turnCostsInSeconds = (long) calcTurnWeight(prevOrNextEdgeId, edgeState.getBaseNode(), edgeState.getEdge());
 
-        return millis + turnCostsInMillis;
+        return millis + 1000 * turnCostsInSeconds;
```

Comment:
Here we have to convert the turn costs to milliseconds to make the turn cost time calculation consistent with the weight calculation.

Reply:
Maybe we should decide before going more public with turn costs to make a breaking change and use the same unit for turn costs? I think I stumbled over the same problem somewhere.

Reply:
You mean store the turn costs as milliseconds ? In this PR or is this just a reminder for later ? Storing turn costs as milliseconds seems a bit extreme, seconds should be enough ? Maybe use 0.1 seconds ?

Reply:
> You mean store the turn costs as milliseconds ?

I would keep storing them as seconds (and later use the same EncodedValue classes to define a different precision), but shouldn't we use the same unit `ms` in the Java API `turnCostEncoder.getTurnCost`? Furthermore to make it more clear maybe we should rename `calcTurnWeight` into `calcTurnCost`? Or even introduce a separate method `calcTurnMillis` like we have for normal weighting ... but are there really cases where a priority to avoid a turn is different to the turn time?

> In this PR or is this just a reminder for later?

Good question - probably later :)

Reply:
> but shouldn't we use the same unit ms in the Java API turnCostEncoder.getTurnCost

The way I understood it so far is that the turn costs do not necessarily have to be turn *times*, maybe someone is using an advanced weighting and drivers just do not want to take turns so each turn adds some penalty (but not necessarily time) to some advanced weight function ? 

> Furthermore to make it more clear maybe we should rename calcTurnWeight into calcTurnCost?

Hmm yeah makes sense because we call it turn cost everywhere else, then again `calcTurnWeight` fits `calcWeight` in the `TurnWeighting` class.. no idea.

> but are there really cases where a priority to avoid a turn is different to the turn time

Hmm I could imagine this, but don't know if anybody does this :)

Reply:
> Hmm I could imagine this, but don't know if anybody does this :)

But which use case do you have in mind where "time != weight"?

The only thing that comes to my mind is an efficient storage. E.g. only two bits are stored and the 0 maps to no turn time, the 1 maps to 5 seconds, the 2 maps to 10 seconds and the 3 maps to infinity (turn restriction). But once we would introduce something like the MappedDecimalEncodedValue I do not see a reason to use the same `ms` as unit.

Maybe there are already users of this more generic turn cost feature, but if we document this change this is no problem IMO. And we should not hesitate to make this API more consistent before it is used more widespread.

BTW: As the defaultUTurnCost is in seconds - is this correct?

Reply:
see #1590